src/machine.rs

```diff
@@ -78,15 +78,21 @@ pub struct MemoryExtra {
 
     /// Whether to enforce the validity invariant.
     pub(crate) validate: bool,
+    /// Contains the `environ` static.
+    pub(crate) environ: Option<Allocation<Tag, AllocExtra>>,
 }
 
 impl MemoryExtra {
-    pub fn new(rng: StdRng, validate: bool, tracked_pointer_tag: Option<PtrId>) -> Self {
+    pub fn new(
+        rng: StdRng, validate: bool,
```

Comment:
That's some odd formatting... either all arguments should be on one line, or they should all have their own line.

Reply:
yes, I have rustfmt disabled to avoid huge diffs. I'll do the formatting when I have a more concise PR